src/parser.yy

```diff
@@ -96,11 +97,18 @@ void InsertToSpecieMap(speciesRatios &ratio, SpeciesPair &toInsert) {
 
 modules  : module
          | modules module
+         | functions
          ;
 
 module : T_DMODULE "name" "{" properties "}" { drv.currentModule.name = $2; drv.FinishParsingModule(); }
 	   ;
 
+functions : function
```

Comment:
Maybe you don't need to have the `functions` non-terminal at all in the grammar. After all, it only exists to implement left-recursion, and that is already implemented in `modules`.

Reply:
bison get's angry if it is non-terminal :S

Reply:
You can't just add the rule `T_DFUNCTION "name" "{" properties "}" { drv.currentModule.name = $2; drv.FinishParsingFunction(); }` to module?

Reply:
i had not thought of this, since i thought they had to be distinct from one another, but the token makes them distinct